src/main/scala/scala/async/internal/TransformUtils.scala

```diff
@@ -133,8 +133,13 @@ private[async] trait TransformUtils {
     else if (fun.tpe == null) (x, y) => false
     else {
       val paramLists = fun.tpe.paramLists
-      val byNamess = paramLists.map(_.map(_.asTerm.isByNameParam))
-      (i, j) => util.Try(byNamess(i)(j)).getOrElse(false)
+      val byNamess: immutable.Seq[mutable.BitSet] = paramLists.map {
+        ps =>
+          val result = new mutable.BitSet()
+          ps.zipWithIndex.foreach { case (p, i) => if (p.asTerm.isByNameParam) result += i }
```

Comment:
May I suggest to avoid zipWithIndex to not allocate the intermediate Tuple2's?
Since the Bitset is never leaked, you could also reuse an empty instance for all cases where there are no by name params (if that is expected to be common).

Reply:
As long as we're going there, I thought paramLists was itself a `List`, which isn't an ideal choice for something you're to index into (unless it's nor performance-critical, which maybe it isn't).  If it were critical I'd short-circuit the no-byname case entirely:

```
var pls = paramLists
var pln = 0
var ps: List[Symbol] = Nil
var pn = 0
var found = false
while (pls.nonEmpty && !found) {
  ps = pls.head
  pn = 0
  while (ps.nonEmpty && !found) {
    p = ps.head
    found = p.asTerm.isByNameParam
    pn += 1
    ps = ps.tail
  }
  pln += 1
  pls = pls.tail
}
if (!found) (i, j) => false
else {
  val ans = new Array[mutable.BitSet](pls.length)
  // Fill in array with logic similar to the above
}
```
  
This would be quite a lot faster (unless `p.asTerm.isByNameParam` is really slow) to create and to use.  But it's a mess of low-level high-performance code, so not a good idea unless this is some sort of bottleneck.